### PR TITLE
sink(ticdc): use validateAndAdjustBound in redo log worker

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/8132

### What is changed and how it works?
- use validateAndAdjustBound in redo log worker
- add test for worker and advancer

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/redo_log_advancer.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+					zap.String("changefeed", a.task.tableSink.changefeed.ID),
+					zap.Stringer("span", &a.task.span),
+					zap.Uint64("memory", refundMem))
+			}
+		}
+		if err := a.redoDMLManager.EmitRowChangedEvents(ctx, a.task.span, releaseMem,
+			a.events...); err != nil {
+			return errors.Trace(err)
+		}
+		a.events = a.events[:0]
```
Comment (hicqu) [inline cdc/processor/sinkmanager/redo_log_advancer.go:97]: Should always re-create a buffer because `EmitRowChangedEvents` is asynchronous.

**Review by hicqu — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**